java/client/src/main/java/glide/api/BaseClient.java

```diff
@@ -1071,7 +1073,29 @@ public CompletableFuture<Map<String, Double>> zrangeWithScores(
     @Override
     public CompletableFuture<Map<String, Double>> zrangeWithScores(
             @NonNull String key, @NonNull ScoredRangeQuery rangeQuery) {
-        return this.zrangeWithScores(key, rangeQuery, false);
+        return zrangeWithScores(key, rangeQuery, false);
+    }
+
+    @Override
+    public CompletableFuture<Object[]> bzmpop(
+            double timeout, @NonNull String[] keys, @NonNull ScoreModifier modifier) {
```

Comment:
Can we make keys come before timeout?  
Generally, we put all the options together and then we could get away with a single function signature.

Reply:
I also dislike it, but in redis api, timeout is the first argument - we follow that
in `bzpopmax` it is the last...

Reply:
we can totally reorder them.

Reply:
Done